src/Microsoft.Azure.EventHubs/EventHubClient.cs

```diff
@@ -599,10 +580,9 @@ internal EventDataSender CreateEventSender(string partitionId = null)
         /// </summary>
         protected override void OnRetryPolicyUpdate()
         {
-            // Propagate retry policy updates to inner sender if there is any.
-            if (this.innerSender != null)
+            if (this.innerSender.IsValueCreated)
```

Comment:
>innerSender [](start = 21, length = 11)

Can you add null check? Tests are failing with NullRef here.

Reply:
ok